src/lib/services/group-service.ts

```diff
@@ -1,39 +1,66 @@
-import { IGroup } from '../types/group';
+import { IGroup, IGroupModel } from '../types/group';
 import { IUnleashConfig, IUnleashStores } from '../types';
-import { ICreateGroup, IGroupStore } from '../types/stores/group-store';
+import { IGroupStore } from '../types/stores/group-store';
 import { Logger } from '../logger';
 import BadDataError from '../error/bad-data-error';
 import { GROUP_CREATED } from '../types/events';
 import { IEventStore } from '../types/stores/event-store';
 import NameExistsError from '../error/name-exists-error';
+import { IUserStore } from '../types/stores/user-store';
 
 export class GroupService {
-    private store: IGroupStore;
+    private groupStore: IGroupStore;
 
     private eventStore: IEventStore;
 
+    private userStore: IUserStore;
+
     private logger: Logger;
 
     constructor(
-        stores: Pick<IUnleashStores, 'groupStore' | 'eventStore'>,
+        stores: Pick<IUnleashStores, 'groupStore' | 'eventStore' | 'userStore'>,
         { getLogger }: Pick<IUnleashConfig, 'getLogger' | 'authentication'>,
     ) {
         this.logger = getLogger('service/group-service.js');
-        this.store = stores.groupStore;
+        this.groupStore = stores.groupStore;
         this.eventStore = stores.eventStore;
+        this.userStore = stores.userStore;
     }
 
-    async getAll(): Promise<IGroup[]> {
-        const groups = await this.store.getAll();
-        return groups;
+    async getAll(): Promise<IGroupModel[]> {
```

Comment:
Seems like a lot of work. Feel free to suggest optimizations.

Reply:
Yeah agreed, I got nothing for now but if this has tests that cover its expected usecases then I don't see any harm with accepting this as is for now and redesigning it a little next week.

I want to say that we could make a query that returns users in a group and replace lines 32-37 with that, which should also reduce the amount of map/reduce/filter here but it's early and I haven't had enough coffee to have thought that through completely